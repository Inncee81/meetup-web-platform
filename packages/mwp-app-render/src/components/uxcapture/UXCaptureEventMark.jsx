// @flow
import * as React from 'react';

type Props = {
	browserEvent: string,
	mark: string,
	children: React$Element<*>,
};

const generateUXCaptureJS = (mark: string) =>
	`if(window.UX) { UX.mark('${mark}'); }`;

/**
 * takes only a *single* child element
 * and injects event based UX.mark() call into the child
 *
 * @see example https://github.com/meetup/ux-capture#image-elements
 */
export default ({ browserEvent, mark, children }: Props) =>
	React.cloneElement(children, {
		[browserEvent]: generateUXCaptureJS(mark),
	});
