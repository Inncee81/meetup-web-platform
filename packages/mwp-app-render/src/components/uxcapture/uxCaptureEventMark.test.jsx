import React from 'react';
import { shallow } from 'enzyme';
import UXCaptureEventMark from './UXCaptureEventMark';

const MOCK_PROPS = {
	mark: 'ux-image-load-logo',
};

const MOCK_CHILD = <img src="/foo" />;

const renderComponent = (props = MOCK_PROPS, children = MOCK_CHILD) =>
	shallow(
		<UXCaptureEventMark {...props}>
			{children}
		</UXCaptureEventMark>
	);

describe('UXCaptureEventMark', () => {
	it('renders component markup', () => {
		global.window = { UX: { mark: jest.fn() } };
		expect(renderComponent()).toMatchSnapshot();
		delete global.window;
	});

	it('returns the child element untouched if it has an onLoad prop', () => {
		global.window = { UX: { mark: jest.fn() } };
		const onLoadChild = <img src="/foo" onLoad="alert('hi');" />;
		expect(renderComponent(MOCK_PROPS, onLoadChild)).toMatchSnapshot();
		delete global.window;
	});
});
