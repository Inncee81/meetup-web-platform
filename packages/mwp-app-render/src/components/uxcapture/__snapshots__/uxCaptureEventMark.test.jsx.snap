// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UXCaptureEventMark renders component markup 1`] = `
<img
  onLoad="if(window.UX) { UX.mark('ux-image-load-logo'); }"
  src="/foo"
/>
`;
